core/store/orm/config.go

```diff
@@ -961,14 +957,30 @@ func parseLogLevel(str string) (interface{}, error) {
 	return lvl, err
 }
 
-func parseUint16(str string) (interface{}, error) {
-	d, err := strconv.ParseUint(str, 10, 16)
-	return uint16(d), err
-}
+var (
+	uintParsers = map[int]func(s string) (interface{}, error){}
+)
 
-func parseUint8(str string) (interface{}, error) {
-	d, err := strconv.ParseUint(str, 10, 8)
-	return uint8(d), err
+func init() {
+	for _, bitSize := range []int{8, 16, 32, 64} {
```

Comment:
This seems a little too clever and neatly escapes type safety, personally I'd prefer just to write out `parseUint8` etc.

Reply:
I don't think I follow where the type safety loss is?

Reply:
`interface{}` and `uintParsers[15]` will panic

Reply:
Hmm yeah I was thinking we were already type unsafe with the hard type assertions i.e. (```parseUint16.(uint32)``` would panic), so it didn't seem like a "loss"

Reply:
Its still more opportunity for failure though, I'll find a better way

Reply:
I concur with @j16r I have no problem with writing out the functions by hand here

Reply:
fixed back to hand rolling